URBNCreativeTestRig/TextDetailTogglesViewController.swift

```diff
@@ -0,0 +1,355 @@
+                label.font = UIFont.boldSystemFontOfSize(14.0)
+                label.translatesAutoresizingMaskIntoConstraints = false
+                label.setContentHuggingPriority(UILayoutPriorityRequired, forAxis: .Horizontal)
+                self.view.addSubview(label)
+            }
+        }
+        
+        var values: Dictionary<String, UIView>= [:]
+        
+        let picker = UIPickerView()
+        picker.dataSource = self
+        picker.delegate = self
+        
+        fontField.text = options.typeFace
+        values["fontField"] = fontField
+        fontField.setTextBlock = {
+            self.options.typeFace = self.fontField.text!
+        }
+
+        sizeField.text = options.fontSize.description
+        sizeField.userInteractionEnabled = false
+        values["sizeField"] = sizeField
+        
+        let sizeStepper = UIStepper()
+        sizeStepper.minimumValue = 0.0
+        sizeStepper.maximumValue = 100.0
+        sizeStepper.stepValue = 1.0
+        sizeStepper.value = options.fontSize
+        sizeStepper.translatesAutoresizingMaskIntoConstraints = false
+        sizeStepper.addTarget(self, action: "fontSizeStepperValueChanged:", forControlEvents: .ValueChanged)
+        self.view.addSubview(sizeStepper)
+        
+        kerningField.text = options.kerning.description
+        kerningField.userInteractionEnabled = false
+        values["kerningField"] = kerningField
+        
+        let kerningStepper = UIStepper()
+        kerningStepper.minimumValue = 0.0
+        kerningStepper.maximumValue = 10.0
+        kerningStepper.stepValue = 0.1
+        kerningStepper.value = options.kerning
+        kerningStepper.translatesAutoresizingMaskIntoConstraints = false
+        kerningStepper.addTarget(self, action: "kerningStepperValueChanged:", forControlEvents: .ValueChanged)
+        self.view.addSubview(kerningStepper)
+        
+        let colorSwatch = UIButton(type: .Custom)
+        colorSwatch.backgroundColor = UIColor(hexString: options.textColor)
+        colorSwatch.translatesAutoresizingMaskIntoConstraints = false
+        colorSwatch.layer.cornerRadius = 15.0
+        colorSwatch.layer.borderColor = UIColor.grayColor().CGColor
+        colorSwatch.layer.borderWidth = 1.0
+        colorSwatch.addTarget(self, action: "colorSwatchTouched", forControlEvents: .TouchUpInside)
+        self.view.addSubview(colorSwatch)
+        
+        colorField.text = options.textColor
+        colorField.delegate = self
+        colorField.setTextBlock = {
+            self.options.textColor = self.colorField.text!
+            colorSwatch.backgroundColor = UIColor(hexString: self.colorField.text)
+        }
+        values["colorField"] = colorField
+        
+        alignmentField.text = options.alignment.rawValue
+        alignmentField.inputView = picker
+        alignmentField.setTextBlock = {
+            self.options.alignment = TextAlignment(rawValue: self.alignmentField.text!)!
```

Comment:
as per Apple's suggestion, use `[unowned self]` when you can guarantee that the objects referencing each other will be deallocated at the same time.